client/web/src/insights/pages/search-insight-creation/CreationSearchInsightPage.tsx

```diff
@@ -0,0 +1,139 @@
+
+    const handleSubmit = useCallback<CreationSearchInsightFormProps['onSubmit']>(
+        async values => {
+            if (!authenticatedUser) {
+                return
+            }
+
+            const {
+                id: userID,
+                organizations: { nodes: orgs },
+            } = authenticatedUser
+            const subjectID =
+                values.visibility === 'personal'
+                    ? userID
+                    : // TODO [VK] Add org picker in creation UI and not just pick first organization
+                      orgs[0].id
+
+            try {
+                const settings = await getSubjectSettings(subjectID).toPromise()
+
+                const newSettingsString = {
+                    title: values.title,
+                    repositories: values.repositories.trim().split(/\s*,\s*/),
+                    series: values.series.map(line => ({
+                        name: line.name,
+                        // Query field is a reg exp field for code insight query setting
+                        // Native html input element adds escape symbols by itself
+                        // to prevent this behavior below we replace double escaping
+                        // with just one series of escape characters e.g. - //
+                        query: line.query.replace(/\\\\/g, '\\'),
+                        stroke: line.color,
+                    })),
+                    step: {
+                        [values.step]: +values.stepValue,
+                    },
+                }
+
+                const edits = jsonc.modify(
+                    settings.contents,
+                    // According to our naming convention <type>.insight.<name>
+                    [`searchInsights.insight.${camelCase(values.title)}`],
+                    newSettingsString,
+                    { formattingOptions: defaultFormattingOptions }
+                )
+
+                const editedSettings = jsonc.applyEdits(settings.contents, edits)
+
+                await updateSubjectSettings(platformContext, subjectID, editedSettings).toPromise()
+
+                history.push('/insights')
+            } catch (error) {
+                return { [FORM_ERROR]: asError(error) }
+            }
+
+            return
+        },
+        [history, updateSubjectSettings, getSubjectSettings, platformContext, authenticatedUser]
+    )
+
+    const handleCancel = useCallback(() => {
+        history.push('/insights')
+    }, [history])
+
+    if (authenticatedUser === null) {
+        return <Redirect to="/" />
+    }
+
+    return (
+        <Page className="col-8">
+            <PageTitle title="Create new code insight" />
+
+            <div className="mb-5">
+                <h2>Create new code insight</h2>
+
+                <p className="text-muted">
+                    Search-based code insights analyse your code based on any search query.{' '}
```

Comment:
```suggestion
                    Search-based code insights analyze your code based on any search query.{' '}
```

cc @AlicjaSuska I see you trying to sneak in your euro-centric spelling :)